recipes/libpng/all/conanfile.py

```diff
@@ -70,6 +72,8 @@ def _configure_cmake(self):
                 cmake.definitions["PNG_ARM_NEON"] = "on"
         if self.options.api_prefix:
             cmake.definitions["PNG_PREFIX"] = self.options.api_prefix
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "aarch64"
```

Comment:
Not only Macos

Reply:
Well, in fact, this is only required when cross-building (because we are not using toolchain file)

Reply:
iOS has different architectures (armv7, armv8 and their variants), so CMAKE_SYSTEM_PROCESSOR will have different values

Reply:
Sure, my suggestion is to _force_ this value only for the cross-building scenario, otherwise, in a native build I would expect it to work out of the box. wdyt?

Reply:
as I understand, in native builds `CMAKE_SYSTEM_PROCESSOR` defaults to `CMAKE_HOST_SYSTEM_PROCESSOR` which should be auto-detected to `aarch64` by cmake itself? still, I don't have ARM M1 laptop to check it, so no idea if it works without explicit definition...

Reply:
I changed to:

```

        if tools.is_apple_os(self.settings.os):
            cmake.definitions["PNG_ARM_NEON"] = "on" if self.settings.os == "Macos" else "off"
            if self.settings.arch == "armv8":
                if self.settings.os == "Macos":
                    cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "aarch64"
                else:
                    cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "arm64"
```

I would expect conan to handle this automatically (except `PNG_ARM_NEON` of course, here we should switch to autotools, it seems to properly handle neon for iOS).